src/server/browserType.ts

```diff
@@ -186,7 +186,8 @@ export abstract class BrowserType extends SdkObject {
       await validateHostRequirements(this._registry, options.channel as registry.BrowserName);
 
     let wsEndpointCallback: ((wsEndpoint: string) => void) | undefined;
-    const wsEndpoint = options.useWebSocket ? new Promise<string>(f => wsEndpointCallback = f) : undefined;
+    const connectOverWebSocket = options.useWebSocket || options.args?.some(a => a.startsWith('--remote-debugging-port='));
```

Comment:
I don't think we actually want to use the websocket connection in this case. Even the current "useWebsocket" is a test-only hack.

Reply:
Discussed offline: If the argument is passed, we will wait that it's listening but not use it.